packages/@uppy/aws-s3/src/index.ts

```diff
@@ -142,8 +142,15 @@ export interface AwsS3Part {
   ETag?: string
 }
 
-type AWSS3WithCompanion = {
-  companionUrl: string
+type AWSS3WithCompanion = (
+  | {
+      /** @deprecated use `endpoint` instead */
```

Comment:
I'm okay with doing a breaking change. If not now then when?

Reply:
On the next major